### PR TITLE
[WFLY-15092] Use maven-resource-plugin filtering to avoid duplicating …

### PR DESCRIPTION
**Pull request by bstansberry**
…module.xml files

https://issues.redhat.com/browse/WFLY-15092

### CODE REVIEW

File: ee-feature-pack/common/src/main/resources/modules/system/layers/base/com/fasterxml/jackson/jaxrs/jackson-jaxrs-json-provider/main/module.xml [hunk 1]
```diff
@@ -23,9 +23,10 @@
   -->
 <module xmlns="urn:jboss:module:1.9" name="com.fasterxml.jackson.jaxrs.jackson-jaxrs-json-provider">
      <resources>
-        <artifact name="${com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider}"/>
-        <artifact name="${com.fasterxml.jackson.jaxrs:jackson-jaxrs-base}"/>
-        <artifact name="${com.fasterxml.jackson.module:jackson-module-jaxb-annotations}"/>
+        <!-- Use filter expressions so standard WildFly and WildFly Preview can use artifacts with different qualifiers -->
+        <artifact name="@gav.exp.start@com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider@module.jakarta.classifier@@gav.exp.end@"/>
+        <artifact name="@gav.exp.start@com.fasterxml.jackson.jaxrs:jackson-jaxrs-base@module.jakarta.classifier@@gav.exp.end@"/>
+        <artifact name="@gav.exp.start@com.fasterxml.jackson.module:jackson-module-jaxb-annotations@module.jakarta.classifier@@gav.exp.end@"/>
```
Comment (yersan) [inline ee-feature-pack/common/src/main/resources/modules/system/layers/base/com/fasterxml/jackson/jaxrs/jackson-jaxrs-json-provider/main/module.xml:29]: A possible alternative to allow the nested expressions avoiding the uses of @gav.exp.start@ and @gav.exp.end@ is to configure a scape character on the maven resource plugin. If we scape the first $, then it seems we could use nested expressions like the following one:

`<artifact name="\${com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider${module.jakarta.classifier}}"/>`

The maven resource plugin has to include the following configuration:

```
<plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-resources-plugin</artifactId>
        <version>3.2.0</version>
        <configuration>
          ...
          <escapeString>\</escapeString>
          ...
        </configuration>
</plugin>
```

Comment (bstansberry) [reply ee-feature-pack/common/src/main/resources/modules/system/layers/base/com/fasterxml/jackson/jaxrs/jackson-jaxrs-json-provider/main/module.xml]: Thanks @yersan ; I've updated to use that.

**Comment by bstansberry** [comment]
WF Preview tests:

https://ci.wildfly.org/viewLog.html?buildId=267256&tab=buildResultsDiv&buildTypeId=WF_WildFlyPreviewLinuxJdk11
https://ci.wildfly.org/viewLog.html?buildId=267254&tab=buildResultsDiv&buildTypeId=WF_Ee9BootableJarLinuxJdk8

**Comment by bstansberry** [comment]
@jamezp or @ronsigal Is this approach to configuring this module ok with you?

**Review by jamezp — APPROVED** [review]

**Comment by jamezp** [comment]
> @jamezp or @ronsigal Is this approach to configuring this module ok with you?

@bstansberry looks good to me.

**Pull request merged**